### PR TITLE
Add support for DPS for PnP samples

### PR DESCRIPTION
**Pull request by bikamani**
TBD:

Change location of samples

### CODE REVIEW

File: device/iot-device-samples/pnp-device-sample/thermostat-device-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/device/Thermostat.java [hunk 1]
```diff
@@ -98,6 +168,60 @@ public void run() {
...
+            if (provisioningStatus.provisioningDeviceClientRegistrationInfoClient.getProvisioningDeviceClientStatus() == ProvisioningDeviceClientStatus.PROVISIONING_DEVICE_STATUS_ERROR ||
+                    provisioningStatus.provisioningDeviceClientRegistrationInfoClient.getProvisioningDeviceClientStatus() == ProvisioningDeviceClientStatus.PROVISIONING_DEVICE_STATUS_DISABLED ||
+                    provisioningStatus.provisioningDeviceClientRegistrationInfoClient.getProvisioningDeviceClientStatus() == ProvisioningDeviceClientStatus.PROVISIONING_DEVICE_STATUS_FAILED)
+            {
+                provisioningStatus.exception.printStackTrace();
+                System.out.println("Registration error, bailing out");
+                break;
+            }
+            System.out.println("Waiting for Provisioning Service to register");
+            Thread.sleep(MAX_TIME_TO_WAIT_FOR_REGISTRATION);
```
Comment (timtay-microsoft) [inline device/iot-device-samples/pnp-device-sample/thermostat-device-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/device/Thermostat.java:194]: No need to sleep 10 seconds each time. 1 second should be fine. Also rename this to POLLING_PERIOD or something similar

File: device/iot-device-samples/pnp-device-sample/thermostat-device-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/device/Thermostat.java [hunk 2]
```diff
@@ -98,6 +168,60 @@ public void run() {
...
+        ClientOptions options = new ClientOptions();
+        options.setModelId(MODEL_ID);
+
+        if (provisioningStatus.provisioningDeviceClientRegistrationInfoClient.getProvisioningDeviceClientStatus() == ProvisioningDeviceClientStatus.PROVISIONING_DEVICE_STATUS_ASSIGNED) {
+            System.out.println("IotHUb Uri : " + provisioningStatus.provisioningDeviceClientRegistrationInfoClient.getIothubUri());
+            System.out.println("Device ID : " + provisioningStatus.provisioningDeviceClientRegistrationInfoClient.getDeviceId());
+
+            String iotHubUri = provisioningStatus.provisioningDeviceClientRegistrationInfoClient.getIothubUri();
+            String deviceId = provisioningStatus.provisioningDeviceClientRegistrationInfoClient.getDeviceId();
+
```
Comment (timtay-microsoft) [inline device/iot-device-samples/pnp-device-sample/thermostat-device-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/device/Thermostat.java:206]: Add a log statement to the effect of "Opening device client..."

**Review by timtay-microsoft — APPROVED** [review]
Just a few small comments, but otherwise looks good

**Review by vinagesh — APPROVED** [review]

**Pull request merged**